### ISSUE TITLE
Create a "CSS Style Guide" for core & plugin developers

### ISSUE THREAD

**Issue by tzi** [piwik/piwik#7652]
Hi piwik team!

As discussed in #7390, it could be urseful to have a CSS style guide for the developers.

The purpose of this minisite is to document our CSS code base, especially the reusable components:

 * It shows the "CSS API" usable by the developers
 * It shows the components that should be restyled in themes
 * It could also used by our visual non-regression testing tool.

It could start with a simple page, and be improved step by step.
There is also some tools to do it like [Style Guide Boilerplate](http://brettjankord.com/projects/style-guide-boilerplate/), [Pattern Primer](https://github.com/adactio/Pattern-Primer) or [Module Gallery](http://modulegallery.keithwyland.com/).

Cheers,
Thomas.

**Comment by mnapoli** [comment]
PR: #7787

**Comment by mnapoli** [comment]
#7787 has been merged but it only includes a UI test of the demo page. This page is not yet documented, and the CSS classes are not tagged as API yet. Let's discuss that topic here:

- do we consider as API everything explained [in this page](https://github.com/piwik/piwik-ui-tests/blob/master/Morpheus_load.png)?
- if yes, we need to integrate this into the developer documentation (via widget/iframe maybe…)

**Comment by tsteur** [comment]
> do we consider as API everything documented in this page?

I do not really have a huge problem with considering everything API but would prefer not to. Eg we have a PHP & JS API for notifications so developers do not really need to know the markup for notifications. Not having it API would just leave us more flexibility in changing things in the future as we will want to change things in the future again for sure. Eg maybe we want to use another UI framework (based on AngularJS, eg https://material.angularjs.org/latest/#/demo/material.components.autocomplete only an example) and there notifications etc are made completely different. It's rather a random example though. I'm ok with it as long as we're careful what we put on that page.

**Comment by mattab** [comment]
Could this issue be closed for 2.14.0? if not, let's move it to 3.0.0 and write here the next step / decision.

**Comment by mnapoli** [comment]
This issue isn't done, I'm moving it to 3.0.

**Comment by mattab** [comment]
we now have a "Demo page" listing CSS styles that are "Api" so the remaining step is to advertise this publicly on developer.piwik.org

**Comment by tzi** [comment]
Hi! Is this page public yet?

**Comment by tsteur** [comment]
Nope, not yet. Need to find a way to make it public somehow. Currently the page needs some admin access + enabled development mode
